pkg/yurthub/cachemanager/cache_manager.go

```diff
@@ -437,14 +457,18 @@ func (cm *cacheManager) saveWatchObject(ctx context.Context, info *apirequest.Re
 				err = cm.storage.Delete(key)
 				delObjCnt++
 			default:
-				// impossible go to here
+				// impossible go here
 			}
 
 			if info.Resource == "pods" {
 				klog.V(2).Infof("pod(%s) is %s", key.Key(), string(watchType))
 			}
 
 			if err != nil {
+				cm.Lock()
+				cm.errorKeys[keyBuildInfo] = err
+				cm.Unlock()
+				cm.deltaFIFO <- Delta{Key: keyBuildInfo, Err: err, Type: Added}
```

Comment:
I worried about the overflow of this channel if there are too many error happened. should we need to consider this corner case?